eval/inception/score_and_FID.py

```diff
@@ -0,0 +1,302 @@
+            
+    Stable version by Dougal J. Sutherland.
+    Params:
+    -- mu1 : Numpy array containing the activations of the pool_3 layer of the
+             inception net ( like returned by the function 'get_predictions')
+             for generated samples.
+    -- mu2   : The sample mean over activations of the pool_3 layer, precalcualted
+               on an representive data set.
+    -- sigma1: The covariance matrix over activations of the pool_3 layer for
+               generated samples.
+    -- sigma2: The covariance matrix over activations of the pool_3 layer,
+               precalcualted on an representive data set.
+    Returns:
+    --   : The Frechet Distance.
+    """
+
+    mu1 = np.atleast_1d(mu1)
+    mu2 = np.atleast_1d(mu2)
+
+    sigma1 = np.atleast_2d(sigma1)
+    sigma2 = np.atleast_2d(sigma2)
+
+    assert mu1.shape == mu2.shape, "Training and test mean vectors have different lengths"
+    assert sigma1.shape == sigma2.shape, "Training and test covariances have different dimensions"
+
+    diff = mu1 - mu2
+
+    # product might be almost singular
+    covmean, _ = linalg.sqrtm(sigma1.dot(sigma2), disp=False)
+    if not np.isfinite(covmean).all():
+        msg = "fid calculation produces singular product; adding %s to diagonal of cov estimates" % eps
+        warnings.warn(msg)
+        offset = np.eye(sigma1.shape[0]) * eps
+        covmean = linalg.sqrtm((sigma1 + offset).dot(sigma2 + offset))
+
+    # numerical error might give slight imaginary component
+    if np.iscomplexobj(covmean):
+        if not np.allclose(np.diagonal(covmean).imag, 0, atol=1e-3):
+            m = np.max(np.abs(covmean.imag))
+            raise ValueError("Imaginary component {}".format(m))
+        covmean = covmean.real
+
+    tr_covmean = np.trace(covmean)
+
+    return diff.dot(diff) + np.trace(sigma1) + np.trace(sigma2) - 2 * tr_covmean
+
+
+
+def inception_score(
+        audio_fps_real,
+        audio_fps_fake,
+        k,
+        metagraph_fp,
+        ckpt_fp,
+        batch_size=100,
+        tf_ffmpeg_ext=None,
+        fix_length=False):
+    # Compute IS and FID scores
+    use_tf_ffmpeg = tf_ffmpeg_ext is not None
+    if not use_tf_ffmpeg:
+        from scipy.io.wavfile import read as wavread
+
+    if len(audio_fps_real) % k != 0:
+        raise Exception('Number of audio files ({}) is not divisible by k ({})'.format(len(audio_fps_real), k))
+    group_size = len(audio_fps_real) // k
+
+    assert(len(audio_fps_real)==len(audio_fps_fake))
+
+    # Restore graph
+    graph = tf.Graph()
+    with graph.as_default():
+        saver = tf.train.import_meta_graph(metagraph_fp)
+        
+        if use_tf_ffmpeg:
+            x_fp = tf.placeholder(tf.string, [])
+            x_bin = tf.read_file(x_fp)
+            x_samps = tf.contrib.ffmpeg.decode_audio(x_bin, tf_ffmpeg_ext, 16000, 1)[:, 0]
+    x = graph.get_tensor_by_name('x:0')
+    activations = graph.get_tensor_by_name('classifier/dense/BiasAdd:0')
```

Comment:
It looks like you are taking the logits (a 10-D vector) and compute the FID score on them. Shouldn't this be the activations of one of the earlier pooling layers (e.g. the pool4 layer) as in the original FID score?

However, one issue that would arise if an earlier layer is taken is that the dimensionality of the activations would be too high (it is already 8192 for the pool4 layer) to compute the FID score in a reasonable time. This is because complexity of the matrix square root computation is cubic in the matrix dimensionality.